pkg/runtime/runtime.go

```diff
@@ -1059,10 +1083,34 @@ func (a *DaprRuntime) publishMessageGRPC(msg *pubsub.NewMessage) error {
 	}
 
 	if err != nil {
-		err = fmt.Errorf("error from app while processing pub/sub event: %s", err)
+		errStatus, hasErrStatus := status.FromError(err)
+		if hasErrStatus {
```

Comment:
nit: we do not need nested-if here. 

```
if hasErrStatus && errStatus.Code() == codes.Unimplemented {
}
```

Reply:
I wanted to avoid depending on the order of operations.